src/test/test_sign.py

```diff
@@ -290,6 +290,61 @@ def test_s2c(self):
+        zero, scalar, bad = self.cbufferize([zero_hex, scalar_hex, 'FF' * EC_SCALAR_LEN])
+        out, out_len = make_cbuffer(zero_hex)
+
+        def add(x, y):
+            ret = wally_ec_scalar_add(x, EC_SCALAR_LEN, y, EC_SCALAR_LEN, out, out_len)
+            return ret, wally_hex_from_bytes(out, out_len)[1]
+
+        def sub(x, y):
+            ret = wally_ec_scalar_subtract(x, EC_SCALAR_LEN, y, EC_SCALAR_LEN, out, out_len)
+            return ret, wally_hex_from_bytes(out, out_len)[1]
+
+        def mul(x, y):
+            ret = wally_ec_scalar_multiply(x, EC_SCALAR_LEN, y, EC_SCALAR_LEN, out, out_len)
+            return ret, wally_hex_from_bytes(out, out_len)[1]
+
+        for fn in [add, sub]:
+            self.assertEqual(fn(None, zero), (WALLY_EINVAL, zero_hex))
+            self.assertEqual(fn(zero, None), (WALLY_EINVAL, zero_hex))
+
+        _, negative_hex = sub(zero, scalar)
+        negative = unhexlify(negative_hex)
+
+        self.assertEqual(add(None, zero), (WALLY_EINVAL, zero_hex))
+        self.assertEqual(add(zero, None), (WALLY_EINVAL, zero_hex))
+        self.assertEqual(add(zero, bad), (WALLY_ERROR, zero_hex)) # WALLY_ERROR = scalar not in G
+        self.assertEqual(add(bad, zero), (WALLY_ERROR, zero_hex))
+        self.assertEqual(add(zero, zero), (WALLY_OK, zero_hex))
+        self.assertEqual(add(zero, scalar), (WALLY_OK, scalar_hex))
+        self.assertEqual(add(zero, negative), (WALLY_OK, negative_hex))
+        self.assertEqual(add(negative, zero), (WALLY_OK, negative_hex))
+        self.assertEqual(add(negative, scalar), (WALLY_OK, zero_hex))
+        self.assertEqual(add(scalar, zero), (WALLY_OK, scalar_hex))
+        self.assertEqual(add(scalar, negative), (WALLY_OK, zero_hex))
+
+        self.assertEqual(sub(None, zero), (WALLY_EINVAL, zero_hex))
+        self.assertEqual(sub(zero, None), (WALLY_EINVAL, zero_hex))
+        self.assertEqual(add(zero, bad), (WALLY_ERROR, zero_hex))
+        self.assertEqual(add(bad, zero), (WALLY_ERROR, zero_hex))
+        self.assertEqual(sub(zero, zero), (WALLY_OK, zero_hex))
+        self.assertEqual(sub(zero, scalar), (WALLY_OK, negative_hex))
+        self.assertEqual(sub(zero, negative), (WALLY_OK, scalar_hex))
+        self.assertEqual(sub(negative, zero), (WALLY_OK, negative_hex))
+        self.assertEqual(sub(scalar, zero), (WALLY_OK, scalar_hex))
+        self.assertEqual(sub(scalar, scalar), (WALLY_OK, zero_hex))
+
+        self.assertEqual(mul(None, zero), (WALLY_EINVAL, zero_hex))
+        self.assertEqual(mul(zero, None), (WALLY_EINVAL, zero_hex))
+        self.assertEqual(mul(zero, bad), (WALLY_ERROR, zero_hex))
+        self.assertEqual(mul(bad, zero), (WALLY_ERROR, zero_hex))
+        self.assertEqual(mul(zero, zero), (WALLY_OK, zero_hex))
+        self.assertEqual(mul(zero, scalar), (WALLY_OK, zero_hex))
+        self.assertEqual(mul(zero, negative), (WALLY_OK, zero_hex))
```

Comment:
Please add some tests for `wally_ec_xonly_verify` too.

Reply:
Done